PROTO_tests/tests/series_test.py

```diff
@@ -0,0 +1,188 @@
+            for series in (
+                ak.Series(data=val, index=idx),
+                ak.Series(data=val),
+                ak.Series(val, idx),
+                ak.Series(val),
+                ak.Series((idx, val)),
+            ):
+                assert isinstance(series, ak.Series)
+                assert isinstance(series.index, ak.Index)
+                assert series.to_list() == ans
+
+        with pytest.raises(TypeError):
+            ak.Series(index=idx)
+
+        with pytest.raises(TypeError):
+            ak.Series((ak.arange(3),))
+
+        with pytest.raises(TypeError):
+            ak.Series()
+
+        with pytest.raises(ValueError):
+            ak.Series(data=ak.arange(3), index=ak.arange(6))
+
+    @pytest.mark.parametrize("dtype", INTEGRAL_TYPES)
+    def test_lookup(self, dtype):
+        idx = ak.arange(3, dtype=dtype)
+        for val in idx, ak.array(["A", "B", "C"]):
+            s = ak.Series(data=val, index=ak.arange(3))
+
+            for key in 1, [0, 2], ak.Index([1]), ak.Index([0, 2]):
+                lk = s.locate(key)
+                assert isinstance(lk, ak.Series)
+                key = ak.array(key) if not isinstance(key, int) else key
+                assert (lk.index == s.index[key]).all()
+                assert (lk.values == s.values[key]).all()
+
+            # testing multi-index lookup
+            mi = ak.MultiIndex([idx, idx[::-1]])
+            s = ak.Series(data=val, index=mi)
+            lk = s.locate(mi[0])
+            assert isinstance(lk, ak.Series)
+            assert lk.index.index == mi[0].index
+            assert lk.values[0] == val[0]
+
+            # ensure error with scalar and multi-index
+            with pytest.raises(TypeError):
+                s.locate(0)
+
+            with pytest.raises(TypeError):
+                s.locate([0, 2])
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", NO_STRING)
+    def test_add(self, prob_size, dtype):
+        added = ak.Series(ak.arange(prob_size // 2, dtype=dtype)).add(
+            ak.Series(
+                data=ak.arange(prob_size // 2, prob_size, dtype=dtype),
+                index=ak.arange(prob_size // 2, prob_size),
+            )
+        )
+        assert added.index.to_list() == ak.arange(prob_size).to_list()
+        assert added.values.to_list() == ak.arange(prob_size, dtype=dtype).to_list()
+
+    @pytest.mark.parametrize("prob_size", pytest.prob_size)
+    @pytest.mark.parametrize("dtype", [ak.int64, ak.float64])
+    # @pytest.mark.parametrize("dtype", NUMERICAL_TYPES)
+    def test_topn(self, prob_size, dtype):
+        prob_size_half = prob_size // 2
+        top = ak.Series(ak.arange(prob_size, dtype=dtype)).topn(prob_size_half)
+        assert top.values.to_list() == list(range(prob_size - 1, prob_size_half - 1, -1))
+        assert top.index.to_list() == list(range(prob_size - 1, prob_size_half - 1, -1))
```

Comment:
I do need some help with this test. For `uint` and `bigint` there is an error that shows up on the messaging from the client. It shows up as an unsupported type, but I can not find where they are actually being called in the messaging.

Reply:
@jaketrookman - This is on the server side processing of `argmaxk` used by `topn`. If you look in `src/KExtremeMsg.chpl` at `maxkMsg`, you will see that it is currently only defined for `int64` and `float64`. Thus, that function will need to be updated in order to support `uint` (which PR #2719 appears to handle) and `bigint` or the test updated to not run on the unsupported types.

Reply:
Updated, but as there is no current implementation of `min/max reduce` for type `bigint`. An issue has been created to look into future implementation.